Add Bun glossary entry (#37814)
mdn/content: files/en-us/glossary/bun/index.md

---
title: Bun
slug: Glossary/Bun
page-type: glossary-definition
---

{{GlossarySidebar}}

> **Bun** is a fast, modern {{Glossary("JavaScript")}} runtime environment designed to prioritize performance and developer tooling. It supports {{Glossary("TypeScript")}}, and JSX out of the box.

Reviewer: ```suggestion
**Bun** is a fast and modern {{Glossary("JavaScript")}} runtime environment designed to prioritize performance and developer tooling.
```

Since you are going to list this as a specific feature below anyway

Unlike [Node.js](https://nodejs.org/) and [Deno](https://deno.com/), Bun is built on Apple's [JavaScriptCore](https://trac.webkit.org/wiki/JavaScriptCore). Notable features of Bun include:

> - State-of-the-art JavaScript execution speed in most real-world scenarios.

Reviewer: ```suggestion
- Drop-in Node.js replacement: almost all Node.js programs can run using Bun-equivalent commands.
- State-of-the-art JavaScript execution speed in most real-world scenarios.
```

> - Extremely fast JavaScript and TypeScript execution.
> - Native support for modern JavaScript features like JSX and ES modules.

Reviewer: There's no runtime that doesn't support ES modules (Node's support may have more rough edges wrt CJS/ESM interop but it's fully spec-compliant with clearly documented behavior). JSX is weird to be called a "modern JavaScript feature". How about this:

```suggestion
- State-of-the-art JavaScript execution speed in most real-world scenarios.
- Native support for JavaScript syntax extensions like JSX and TypeScript.
```

Author: That makes more sense :)

> - Global cache for managing package installations to minimize disk usage.

Reviewer: I didn't know this is true. I thought the mechanism is exactly like that of npm. Do you have support materials?

Author: Yes, I read it about it here, all packages downloaded are stored in to a global cache locally, and then they are copied over when installing to the `node_modules` using the appropriate method based on the OS.  https://bun.sh/docs/install/cache.

Reviewer: I see. I'm on mac so I'm not aware that it does symlinking since node_modules looks like regular files to me. Now I understand that it's actually CoW.

> - Built-in bundler, transpiler, [Jest](https://jestjs.io/) compatible test-runner and package manager.

Reviewer: ```suggestion
- Built-in bundler, transpiler, [Jest](https://jestjs.io/) compatible test-runner, package manager, and more built-in utilities that web developers may find useful.
```
